src/Symfony/Component/Security/Core/Authorization/ExpressionLanguage.php

```diff
@@ -13,6 +13,10 @@
 
 use Symfony\Component\ExpressionLanguage\ExpressionLanguage as BaseExpressionLanguage;
 
+if (!class_exists('Symfony\Component\ExpressionLanguage\ExpressionLanguage')) {
+    throw new \RuntimeException('The ExpressionLanguage class requires the "ExpressionLanguage" component. Install "symfony/expression-language" to use it.');
+}
```

Comment:
Unfortunately this approach will not work. The class must be in the "else".

Reply:
Why ? If an exception is thrown PHP will never reach the class declaration and won't try to find the parent class.

Reply:
because PHP parses class ahead of time, we already got bitten by that in the past

Reply:
Yes some things are done ahead of time for class declarations, but in the case we have here it works without having to put the whole class in an else clause : https://3v4l.org/mELj9

Reply:
we already merged such kind of code, and it broke, opcache involved maybe
I wouldn't take the risk another time

Reply:
@nicolas-grekas I've modified my commit in according with your suggestions. I hope is all ok, thank you.